### PR TITLE
consensus: Improve debug-level logging 

### LINKED ISSUE #1004: consensus: Enhance debug level logging details
#### Summary
 By enabling debug level only for dusk_consensus crate we should be capable of capturing any meaningful detail of consensus execution in way that facilitates both debugging and reproducing a misbehavior/error. 

#### Possible solution design or implementation
Info level - basic traces of consensus execution
Debug level - detailed information of any consensus event
Trace level - traces all messages 

Additionally, a  config param or env variable to be added for setting `with_env_filter(..)` externally.

### LINKED ISSUE #1010: consensus: Allow voting on empty block hash
**Describe the bug**
When a block generator does not receive a valid candidate block, then 1st_reduction is initialized with  an empty block. As per the golang-impl, we should allow an extracted provisioner to vote on an empty hash and eventually reach an event of `votes converged for an empty hash`. 

Additionally, verify_block_header must discard any block with empty block hash.

### PR DESCRIPTION
**Pull request by goshawk-3**
fixes #1004 
fixes #1010

### CODE REVIEW

File: consensus/src/agreement/step.rs [hunk 1]
```diff
@@ -220,7 +231,7 @@ impl<D: Database> Executor<D> {
             )
             .await;
 
-            tracing::debug!("broadcast aggr_agreement {:#?}", msg);
+            tracing::trace!("broadcast aggr_agreement {:#?}", msg);
```
Comment (moCello) [inline consensus/src/agreement/step.rs:234]: same for here, consider importing `trace` macro

File: consensus/src/agreement/step.rs [hunk 2]
```diff
@@ -176,6 +178,15 @@ impl<D: Database> Executor<D> {
             return None;
         }
 
+        let hdr = &msg.header;
+        tracing::debug!(
```
Comment (moCello) [inline consensus/src/agreement/step.rs:182]: `debug` is already imported and can be used without `tracing::`

File: consensus/src/execution_ctx.rs [hunk 3]
```diff
@@ -115,7 +115,7 @@ impl<'a> ExecutionCtx<'a> {
                 // Timeout event. Phase could not reach its final goal.
                 // Increase timeout for next execution of this step and move on.
                 Err(_) => {
-                    tracing::info!(event = "timeout");
+                    tracing::info!(event = "timeout-ed");
```
Comment (moCello) [inline consensus/src/execution_ctx.rs:118]: consider removing `tracing::`

**Review by moCello — APPROVED** [review]
AFAICS LGTM
just some tiny nits that you are very free to ignore

File: consensus/src/execution_ctx.rs (hunk 3 above)
Comment (goshawk-3) [reply consensus/src/execution_ctx.rs]: @moCello I removed all occurrences of `tracing::` usage

**Review by moCello — APPROVED** [review]
LGTM

**Pull request merged**